core/modules/menu/menu.module

```diff
@@ -697,15 +697,16 @@ function menu_form_node_form_alter(&$form, $form_state) {
     '#type' => 'textfield',
     '#title' => t('Menu link title'),
     '#maxlength' => 255,
-    '#default_value' => $link['link_title'],
+    '#default_value' => $link['mlid'] ? $link['link_title'] : '',
   );
 
   $form['menu']['link']['description'] = array(
-    '#type' => 'textarea',
-    '#title' => t('Description'),
+    '#type' => 'textfield',
+    '#size' => 120,
+    '#title' => t('Description or clarification'),
     '#default_value' => isset($link['options']['attributes']['title']) ? $link['options']['attributes']['title'] : '',
     '#rows' => 1,
-    '#description' => t('Shown when hovering over the menu link.'),
+    '#description' => t('Shown when hovering over the menu link.') . ' ' . t('Often read by screen readers.'),
```

Comment:
Are these changes necessary for this issue? Sounds like this should be a separate issue, as I imagine there'd be other places that could use these changes too...
Also, I don't think the description text should be split between two `t()` functions. Seems unnecessary and harder to read (code-wise).